pkgs/build-support/rust/default.nix

```diff
@@ -54,9 +61,14 @@ let
   ccForHost="${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc";
   cxxForHost="${stdenv.cc}/bin/${stdenv.cc.targetPrefix}c++";
   releaseDir = "target/${rustTarget}/${buildType}";
+
+  # Fetcher implementation choice should not be part of the hash in final
+  # derivation; only the cargoSha256 input matters.
+  filteredArgs = builtins.removeAttrs args [ "legacyCargoFetcher" ];
```

Comment:
Also added this, which ensures that the default value of `legacyCargoFetcher` does not impact the hash sent to `stdenv.mkDerivation`.  This lets us do the next step (default it to true, opting out all packages via a treewide sed) without causing any hash changes or package rebuilds, including in packages like `cargo` that do use the Rust build platform infra but don't use any fetchers (it inherits its vendor dir from `rustc`).